plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt

```diff
@@ -227,9 +227,13 @@ class CodeModernizerSession(
         } catch (e: Exception) {
             val errorMessage = "Unexpected error when uploading artifact to S3: ${e.localizedMessage}"
             LOG.error { errorMessage }
+
+            state.putJobHistory(sessionContext, TransformationStatus.FAILED)
+            state.currentJobStatus = TransformationStatus.FAILED
+            CodeModernizerStartJobResult.UploadArtifactToS3Failure
```

Comment:
Don't you want to `return` this since you removed the `throw e`?

Reply:
Since return type of this function is a String you won't be able to return this error, so still need to `throw e`

Reply:
Then that is caught here: https://github.com/aws/aws-toolkit-jetbrains/blob/28b1aa5fee49793e45f6b7659867ecba4dbeabcf/plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt#L154

Reply:
So I don't think you need a separate S3 Upload Error type for this.

Just modify message "codemodernizer.notification.warn.unable_to_start_job"

Reply:
Actually, on second thought, that ^ message is also shown when our startJob API fails, so probably do want to handle this case separately.

Reply:
Yeah I tried this way but I need to generalize the error as I dont have choice now. So I will update accordingly.

Reply:
So maybe throw the error when the upload fails with a specific error message, then check the error message in the catch block on line 154 to decide what to return: UnableToUploadToS3 if error messages match and UnableToStartJob otherwise.

Reply:
I think that's fine for now